levelkdev/swapr-dapp
# Fix/static provider

# Summary
This implementation fixes a problem with the provider.
The problem was first noticed while working on curve pools. `getQuote` required us to use the provider and there was a NONETWORK exception thrown. Adding the network to the static provider fixes it.

  # To Test

1) Link the sdk from on `curve/factory-pools-router` or `release/1.6.0-rc.3`
<p><code> rm node_modules </code><br />
<code>npm install</code><br />
<code>npm link </code> </p>
2) Select an appropriate Curve trade (USDT/USDC should work on mainnet/gnosis chains) and set an amount.
<p><code> rm node_modules  </code><br />
<code>yarn link "@swapr/sdk"  </code><br />
<code>yarn install  </code><br />
<code>yarn start </code> </p>
3) While computing the trades, check console logs. This work is built on top of `feature/arbitrum-goerli` from the dapp. In that branch there are some warnings from the Curve trade that the default provider doesn't work. In this branch those logs are no longer present. This means the provider is working as intended.

(Sorry for the branch salad here)
EDIT: Added commands for linking, compiling dependencies and starting the server. 

This work is done on top of `feature/arbitrum-goerli` (now merged). It will be a much smaller change after that PR is merged to develop. The core of the changes are in `src/state/swap/hooks.ts`.
